sharedbuffers/mapped_struct.py

```diff
@@ -781,10 +783,13 @@ def _metadata(self,
             else:
                 raise ValueError("Inconsistent data, unknown type code %r" % (dcode,))
 
-    @cython.locals(offs = cython.ulonglong, idmap = dict)
-    def __init__(self, buf, offs, idmap = None):
+    @cython.locals(offs = cython.ulonglong, idmap = dict, elem_start = cython.ulonglong,
+        elem_end = cython.longlong)
```

Comment:
elem_end should be ulonglong too, mixing them can cause undefined behavior.

Maybe initialize it to ~0 and just consider that it could be greater than the size of the list when operating with it.

Or use a triplet: `slice_start, slice_end, slice_step`, use `slice_step=0` as a flag to not slice at all (the whole sequence).